flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -355,15 +356,51 @@ public void handleGlobalFailure() {
 		assertThat(deployedExecutionVertices, contains(executionVertexId, executionVertexId));
 	}
 
+	@Test
+	public void vertexIsNotAffectedByOutdatedDeployment() {
+		final JobGraph jobGraph = singleJobVertexJobGraph(2);
+
+		testExecutionSlotAllocator.disableAutoCompletePendingRequests();
+		final DefaultScheduler scheduler = createSchedulerAndStartScheduling(jobGraph);
+
+		final Iterator<ArchivedExecutionVertex> vertexIterator = scheduler.requestJob().getAllExecutionVertices().iterator();
+		final ArchivedExecutionVertex v1 = vertexIterator.next();
+		final ArchivedExecutionVertex v2 = vertexIterator.next();
+
+		final SchedulingExecutionVertex sv1 = scheduler.getSchedulingTopology().getVertices().iterator().next();
+
+		// fail v1 and let it recover to SCHEDULED
```

Comment:
maybe add as a comment that the initial deployment of v1 will be outdated

Reply:
Ok. Done.